hazelcast/src/main/java/com/hazelcast/nio/serialization/AbstractGenericRecord.java

```diff
@@ -28,7 +29,7 @@
 /**
  * Implementation of GenericRecord interface to give common equals and hashcode implementation
  */
-public abstract class AbstractGenericRecord implements GenericRecord {
+public abstract class AbstractGenericRecord implements InternalGenericRecord {
```

Comment:
I see that this class is extended in lots of places including the ones we give to the user. Is that necessary? If so, there are some unnecessary `implements InternalGenericRecord` statements on some of the classes that extend this(or one of its childs)

Reply:
When in memory format is Object , the stored object is DeserializedGenericRecord. 
And the query needs to work on it. `GenericRecordQueryReader` excepts only `InternalGenericRecord`. 
Therefore all implementations needs to be `InternalGenericRecord`